src/rrt/Tree.hpp

```diff
@@ -309,7 +350,21 @@ class Tree {
         }
 
         // Add a node to the tree for this state
+        int lengthT = sizeof(intermediateState) / sizeof(0.0);
+        double* data = new double[lengthT];
+        if (NULL == _TToArray) {
+            for (int i = 0; i < lengthT; i++) {
+                data[i] = intermediateState[i];
+            }
+        } else {
+            data = _TToArray(intermediateState);
```

Comment:
Another option is having the double array as an argument instead of the return value. This allows for you to do both the memory construction and deletion instead of having it split across multiple locations